code/modules/events/spacevine.dm

```diff
@@ -9,7 +9,7 @@
 
 	for(var/area/hallway/A in world)
 		for(var/turf/simulated/F in A)
-			if(!F.density && !F.contents.len)
+			if(!F.density || !F.contents.len)

```

Comment:
this will now pick turfs with things on them, like machines and tables and stuff, which looking at the code prior to your change, is not intentional.